### PR TITLE
feat(fiatconnect): add timeout to fiat connect requests

### PR DESCRIPTION
**Pull request by satish-ravi**
### Description

react-native's fetch implementation doesn't have a default timeout option and requests can hang for a long time. This PR uses the latest fiatconnect-sdk (which supports timeouts) to set a timeout on all fiatConnect and IHL requests. The timeout is configurable through firebase remote config

### Other changes

N/A

### Tested

Unit tests, manually by setting a low value for timeout (500ms) and checking if selecting a quote for a provider, which should trigger a login with the provider, errors out with an AbortError .

### How others should test

N/A

### Related issues

- Fixes ACT-499

### Backwards compatibility

Yes

### CODE REVIEW

File: src/fiatconnect/clients.test.tsx [hunk 1]
```diff
@@ -50,6 +51,10 @@ describe('getFiatConnectClient', () => {
     const fcClient = await getFiatConnectClient('provider1', 'https://provider1.url')
     expect(getWalletAsync).toHaveBeenCalled()
     expect(fcClient).toBeInstanceOf(FiatConnectClient)
+    expect(fcClient).toHaveProperty(
+      'config.timeout',
+      REMOTE_CONFIG_VALUES_DEFAULTS.fiatConnectTimeoutSeconds * 1000
+    )
```
Comment (satish-ravi) [inline src/fiatconnect/clients.test.tsx:57]: This test depends on the implementation detail of the FiatConnectClient. Open to suggestions on an easy way to test this without this dependency

File: src/firebase/remoteConfigValuesDefaults.ts [hunk 2]
```diff
@@ -62,6 +62,7 @@ export const REMOTE_CONFIG_VALUES_DEFAULTS: Omit<
   celoWithdrawalEnabledInExchange: true,
   fiatConnectCashInEnabled: false,
   fiatConnectCashOutEnabled: false,
+  fiatConnectTimeoutSeconds: 30,
```
Comment (satish-ravi) [inline src/firebase/remoteConfigValuesDefaults.ts:65]: Was considering 10s or 30s. Saw some mock-provider requests taking 5s, so went with 30s to be safe.

**Review by jeanregisser — APPROVED** [review]
Great! 👍 
We should apply this global to all our fetch calls.

File: src/firebase/remoteConfigValuesDefaults.ts (hunk 2 above)
Comment (jeanregisser) [reply src/firebase/remoteConfigValuesDefaults.ts:65]: yes 30s is good in a mobile environment.

Comment (jeanregisser) [reply src/firebase/remoteConfigValuesDefaults.ts:65]: yes 30s is good in a mobile environment.

File: src/fiatconnect/clients.test.tsx (hunk 1 above)
Comment (kathaypacific) [reply src/fiatconnect/clients.test.tsx:57]: this seems okay to me, you could `jest.mock` the value but i don't see much of a benefit.

**Review by kathaypacific — APPROVED** [review]
LGTM! ✨

**Review by cajubelt — APPROVED** [review]

**Review by valora-bot-crowdin — APPROVED** [review]
looks great!

**Comment by jh2oman** [comment]
^ that bot was me. Forgot to logout 😅

**Pull request merged**